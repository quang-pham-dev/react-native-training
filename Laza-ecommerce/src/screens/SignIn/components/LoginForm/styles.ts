import { Platform, StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';

export const styles = StyleSheet.create({
  main: {
    flex: 2,
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: Metrics.padding.lg,
  },

  inputTitle: {
    alignSelf: 'flex-start',
    marginTop: Metrics.margin.lg,
  },
  input: {
    width: '100%',
    paddingVertical: Metrics.padding.xm,
    borderBottomWidth: 1,
    borderBottomColor: Colors.borderGray,
  },

  errorMessage: {
    alignSelf: 'flex-start',
    fontSize: Fonts.size.small,
    color: Colors.textRed,
  },

  forgotPasswordWrap: {
    marginTop: Metrics.margin.xxl,
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'flex-end',
    marginBottom: Metrics.margin.lg,
  },
  forgotPassword: {
    fontFamily: 'Inter_400Regular',
    fontSize: Fonts.size.normal,
    lineHeight: Fonts.lineHeight.base,
    color: Colors.textRed,
  },

  rememberMeWrap: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: Metrics.margin.lg,
  },
  rememberMeText: {
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    fontSize: Fonts.size.small,
    lineHeight: Fonts.lineHeight.xs,
    color: Colors.textBlack,
  },

  rememberMeSwitch: {
    transform:
      Platform.OS === 'android'
        ? [{ scaleX: 0.9 }, { scaleY: 0.9 }]
        : [{ scaleX: 0.6 }, { scaleY: 0.6 }],
  },

  footer: {
    flex: 1,
    justifyContent: 'flex-end',
  },

  footerTextWrapper: {
    width: '100%',
    flexDirection: 'row',
    flexWrap: 'wrap',
    paddingHorizontal: Metrics.margin.lg,
    marginBottom: Metrics.margin.xl,
  },
  condition: {
    textAlign: 'center',
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    color: Colors.textGray,
    fontSize: Fonts.size.small,
    lineHeight: Fonts.lineHeight.sm,
  },
  Term: {
    color: Colors.textBlack,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
  },

  bottomButton: {
    width: '100%',
    height: 75,
    paddingTop: Metrics.padding.xm,
    alignItems: 'center',
  },
  loginButton: {
    backgroundColor: Colors.primaryColor,
  },
  textBottomButton: {
    fontSize: Fonts.size.default,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    color: Colors.white,
    lineHeight: Fonts.lineHeight.sm,
  },
});

export default styles;
