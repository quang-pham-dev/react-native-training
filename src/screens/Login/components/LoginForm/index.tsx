import React from 'react'

// LIB
import {useForm, Controller, SubmitHandler} from 'react-hook-form'
import * as Yup from 'yup'
import {yupResolver} from '@hookform/resolvers/yup/dist/yup'
import {vs} from 'react-native-size-matters/extend'

// Contexts
import {useAuthContext} from '@contexts/auth/AuthContext'

// Components
import Input, {InputType} from '@components/Input'
import Switch from '@components/Switch'

// Themes
import {Icons, Metrics} from '@themes'

// Styles
import ViewStyled from '@components/View/View.styles'
import PStyled from '@components/Paragraph/P.styles'
import FlexStyled from '@components/Flex/Flex.styles'

// Types
import {LoginCredentials} from '@model-types'
import {PARAGRAPH_TYPE} from '@constants'
import {BtnType, Button} from '@components/Button'

// initial values
const loginFormInit: LoginCredentials = {
  username: '',
  password: '',
}

// validation schema
const loginSchema = Yup.object({
  username: Yup.string().required('Username is required'),
  password: Yup.string()
    .required('Password is required')
    .min(6, 'Password must be at least 6 characters'),
}).required()

interface ILoginFormProps {
  onSubmit: (data: LoginCredentials) => void
}

const LoginForm = ({onSubmit}: ILoginFormProps) => {
  const {state} = useAuthContext()

  const {isLoading} = state || {}

  const {
    control,
    handleSubmit,
    formState: {errors},
  } = useForm<LoginCredentials>({
    mode: 'onChange',
    defaultValues: loginFormInit,
    resolver: yupResolver(loginSchema),
  })

  const onSubmitLoginForm: SubmitHandler<LoginCredentials> = (
    loginInfo: LoginCredentials,
  ) => {
    onSubmit(loginInfo)
  }

  return (
    <>
      <ViewStyled.Custom pHorizontal={Metrics.padding.mediumPlus}>
        {/* Username */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({field: {onChange, value}}) => (
            <Input
              label="Username"
              placeholder="Enter your username"
              icon={Icons.filledProfile}
              value={value}
              type={InputType.NORMAL}
              onChange={(val: string) => onChange(val)}
            />
          )}
          name="username"
        />
        {errors.username && (
          <PStyled.ErrorMessage>
            {errors.username?.message}
          </PStyled.ErrorMessage>
        )}

        {/* Password */}
        <ViewStyled.Custom pTop={vs(Metrics.padding.mediumPlus)}>
          <Controller
            control={control}
            rules={{
              required: true,
            }}
            render={({field: {onChange, value}}) => (
              <Input
                onChange={(val: string) => onChange(val)}
                value={value}
                label="Password"
                placeholder="Enter your password"
                type={InputType.PASSWORD}
              />
            )}
            name="password"
          />
          {errors?.password && (
            <PStyled.ErrorMessage>
              {errors.password?.message}
            </PStyled.ErrorMessage>
          )}
        </ViewStyled.Custom>
        <ViewStyled.Custom pTop={vs(Metrics.margin.extraMedium)}>
          <PStyled.ErrorMessage textAlign="right">
            Forgot password?
          </PStyled.ErrorMessage>
        </ViewStyled.Custom>
        <FlexStyled.FlexSpaceBetweenCenter pTop={vs(Metrics.margin.large)}>
          <PStyled.Base>Remember me</PStyled.Base>
          <Switch />
        </FlexStyled.FlexSpaceBetweenCenter>
      </ViewStyled.Custom>

      <FlexStyled.FlexEnd flex={1}>
        <FlexStyled.RowCenter>
          <PStyled.Center type={PARAGRAPH_TYPE.CONFIRM_AGREE}>
            {'By connecting your account confirm that you agree \nwith our '}
            <PStyled.Base type={PARAGRAPH_TYPE.TERM_AND_CONDITION}>
              {'Term and Condition'}
            </PStyled.Base>
          </PStyled.Center>
        </FlexStyled.RowCenter>

        <ViewStyled.Custom pTop={vs(26)}>
          <Button
            type={BtnType.BOTTOM}
            disabled={!!errors?.username || !!errors?.password}
            onPress={handleSubmit(onSubmitLoginForm)}
            label="Login"
            isLoading={isLoading}
          />
        </ViewStyled.Custom>
      </FlexStyled.FlexEnd>
    </>
  )
}

export default LoginForm
