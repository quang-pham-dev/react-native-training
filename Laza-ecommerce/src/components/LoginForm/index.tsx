import React, { memo, useContext, useState } from 'react';
import { Alert, Switch, Text, View } from 'react-native';
import isEqual from 'react-fast-compare';

// LIB
import { useForm, Controller, SubmitHandler } from 'react-hook-form';
import * as Yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup/dist/yup';

// Components
import Button from 'components/Button';
import TextInput from 'components/TextInput';
import LoadingIndicator from 'components/LoadingIndicator';

// Context
import { AppContext } from 'context/AppContext';
import { SIGN_IN_SUCCESS, SIGN_IN_FAILED, SIGN_IN } from 'context/actions/auth';

// API
import { authService } from 'api';

// Constants
import { AUTH_DATA } from 'constants/Common';

// Utils
import { set } from 'utils/localStorage';

// Types
import { IUserSignIn } from 'types/models/User';
import { LOADING_SIZE } from 'types/common/Enums';

// Styles
import { styles } from './styles';

// initial values
const loginFormInit: IUserSignIn = {
  username: '',
  password: '',
};

// validation schema
const loginSchema = Yup.object({
  username: Yup.string().required('Username is required'),
  password: Yup.string()
    .required('Password is required')
    .min(6, 'Password must be at least 6 characters'),
}).required();

const LoginForm = () => {
  const [isEnabled, setIsEnabled] = useState(true);
  const toggleSwitch = () => setIsEnabled(previousState => !previousState);
  const { authState, authDispatch } = useContext(AppContext);

  const {
    control,
    reset,
    handleSubmit,
    formState: { errors },
  } = useForm<IUserSignIn>({
    mode: 'onChange',
    defaultValues: loginFormInit,
    resolver: yupResolver(loginSchema),
  });

  // handle action call api SignIn when user press Login button
  const handOnSubmit: SubmitHandler<IUserSignIn> = async (loginInfo: IUserSignIn) => {
    authDispatch({ type: SIGN_IN });
    const { username, password } = loginInfo;
    try {
      const response = await authService.signIn(username, password);
      const data = JSON.stringify(response.data);
      // set auth data info to local storage
      await set(AUTH_DATA, data);
      const { user } = response.data;
      const { access_token } = response.data;
      if (user && access_token) {
        authDispatch({
          type: SIGN_IN_SUCCESS,
          payload: {
            user,
            access_token,
          },
        });
      }
    } catch (error) {
      authDispatch({ type: SIGN_IN_FAILED, payload: error });
      Alert.alert('Error', error.message);
      // reset form
      reset(loginFormInit);
    }
  };

  return (
    <>
      {authState?.isProcessing && <LoadingIndicator size={LOADING_SIZE.LARGE} />}
      <View style={styles.main}>
        {/* Username */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Username'
              labelStyle={styles.inputTitle}
              placeholder='Enter your username'
              testID='usernameInput'
            />
          )}
          name='username'
        />
        {errors.username && (
          <Text testID='usernameInputError' accessibilityRole='text' style={styles.errorMessage}>
            {errors.username?.message}
          </Text>
        )}

        {/* Password */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Password'
              labelStyle={styles.inputTitle}
              secureTextEntry
              placeholder='Enter your password'
              testID='passwordInput'
            />
          )}
          name='password'
        />
        {errors.password && (
          <Text testID='passwordInputError' style={styles.errorMessage}>
            {errors.password?.message}
          </Text>
        )}

        <View style={styles.forgotPasswordWrap}>
          <Text style={styles.forgotPassword}>Forgot password?</Text>
        </View>
        <View style={styles.rememberMeWrap}>
          <Text style={styles.rememberMeText}>Remember me</Text>
          <Switch
            style={styles.rememberMeSwitch}
            trackColor={{ false: '#767577', true: '#34C759' }}
            thumbColor={isEnabled ? 'white' : '#f4f3f4'}
            ios_backgroundColor='#3e3e3e'
            onValueChange={toggleSwitch}
            value={isEnabled}
          />
        </View>
      </View>
      <View style={styles.footer}>
        <View style={styles.footerTextWrapper}>
          <Text style={styles.condition}>
            By connecting your account confirm that you agree with our
            <Text style={styles.Term}> Term and Condition</Text>
          </Text>
        </View>

        <Button
          testID='loginButton'
          text='Login'
          buttonStyles={[styles.bottomButton, styles.loginButton]}
          textStyles={[styles.textBottomButton]}
          onPress={handleSubmit(handOnSubmit)}
        />
      </View>
    </>
  );
};

export default memo(LoginForm, isEqual);
