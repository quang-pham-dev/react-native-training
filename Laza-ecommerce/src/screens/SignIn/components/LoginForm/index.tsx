import React, { memo, useState } from 'react';
import { Switch, Text, View } from 'react-native';

// LIB
import isEqual from 'react-fast-compare';
import { useForm, Controller, SubmitHandler } from 'react-hook-form';
import * as Yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup/dist/yup';

// Components
import Button from 'components/Button';
import TextInput from 'components/TextInput';

// Types
import { ILoginCredentials } from 'types/models/User';

// Styles
import { styles } from './styles';
import Colors from 'themes/Colors';

// initial values
const loginFormInit: ILoginCredentials = {
  username: '',
  password: '',
};

// validation schema
const loginSchema = Yup.object({
  username: Yup.string().required('Username is required'),
  password: Yup.string()
    .required('Password is required')
    .min(6, 'Password must be at least 6 characters'),
}).required();

interface ILoginFormProps {
  onSubmit: (data: ILoginCredentials) => void;
}

const LoginForm = ({ onSubmit }: ILoginFormProps) => {
  const [isEnabled, setIsEnabled] = useState(true);
  const toggleSwitch = () => setIsEnabled(previousState => !previousState);
  const {
    control,
    handleSubmit,
    formState: { errors },
  } = useForm<ILoginCredentials>({
    mode: 'onChange',
    defaultValues: loginFormInit,
    resolver: yupResolver(loginSchema),
  });

  const onSubmitHandler: SubmitHandler<ILoginCredentials> = async (
    loginInfo: ILoginCredentials,
  ) => {
    onSubmit(loginInfo);
  };

  return (
    <>
      <View style={styles.main}>
        {/* Username */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Username'
              labelStyle={styles.inputTitle}
              placeholder='Enter your username'
              testID='usernameInput'
            />
          )}
          name='username'
        />
        {errors.username && (
          <Text testID='usernameInputError' accessibilityRole='text' style={styles.errorMessage}>
            {errors.username?.message}
          </Text>
        )}

        {/* Password */}
        <Controller
          control={control}
          rules={{
            required: true,
          }}
          render={({ field: { onChange, value } }) => (
            <TextInput
              onChangeText={(value: string) => onChange(value)}
              value={value}
              textInputStyles={styles.input}
              label='Password'
              labelStyle={styles.inputTitle}
              secureTextEntry
              placeholder='Enter your password'
              testID='passwordInput'
            />
          )}
          name='password'
        />
        {errors?.password && (
          <Text testID='passwordInputError' style={styles.errorMessage}>
            {errors?.password?.message}
          </Text>
        )}

        <View style={styles.forgotPasswordWrap}>
          <Text style={styles.forgotPassword}>Forgot password?</Text>
        </View>
        <View style={styles.rememberMeWrap}>
          <Text style={styles.rememberMeText}>Remember me</Text>
          <Switch
            style={styles.rememberMeSwitch}
            trackColor={{ false: Colors.lightGray, true: Colors.lightGreen }}
            thumbColor={isEnabled ? Colors.white : Colors.lightGray}
            ios_backgroundColor={Colors.lightGray}
            onValueChange={toggleSwitch}
            value={isEnabled}
          />
        </View>
      </View>
      <View style={styles.footer}>
        <View style={styles.footerTextWrapper}>
          <Text style={styles.condition}>
            By connecting your account confirm that you agree with our
            <Text style={styles.Term}> Term and Condition</Text>
          </Text>
        </View>

        <Button
          testID='loginButton'
          text='Login'
          buttonStyles={[styles.bottomButton, styles.loginButton]}
          textStyles={[styles.textBottomButton]}
          onPress={handleSubmit(onSubmitHandler)}
        />
      </View>
    </>
  );
};

export default memo(LoginForm, isEqual);
