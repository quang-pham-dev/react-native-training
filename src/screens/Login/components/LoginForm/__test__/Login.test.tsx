import React from 'react'

// LIBS
import {render} from '@testing-library/react-native'

// Components
import LoginForm from '@screens/Login/components/LoginForm'

// constants

import {LoginCredentials} from '@model-types'

// Types

const mockLogin = jest.fn((data: LoginCredentials) => {
  return Promise.resolve({data})
})

describe('Login Form', () => {
  let tree: any
  beforeEach(() => {
    tree = render(<LoginForm onSubmit={mockLogin} />)
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot()
  })
})
