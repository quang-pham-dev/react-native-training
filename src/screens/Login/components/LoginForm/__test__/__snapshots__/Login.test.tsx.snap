// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form should render correctly 1`] = `
Array [
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingHorizontal": 20,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8F959E",
              "fontFamily": "Inter-Regular",
              "fontSize": 13,
              "lineHeight": 14,
            },
          ]
        }
      >
        Username
      </Text>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderBottomColor": "#E7E8EA",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "width": "100%",
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          keyboardType="default"
          onChangeText={[Function]}
          placeholder="Enter your username"
          secureTextEntry={false}
          style={
            Array [
              Object {
                "color": "#1D1E20",
                "flexBasis": 0,
                "flexGrow": 1,
                "flexShrink": 1,
                "fontFamily": "Inter-Medium",
                "fontSize": 15,
                "lineHeight": 16,
                "paddingLeft": 0,
                "paddingRight": 5,
                "paddingVertical": 15,
              },
            ]
          }
          type="normal"
          underlineColorAndroid="transparent"
          value=""
        />
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/icons/filledProfile.png",
            }
          }
          style={
            Array [
              Object {},
            ]
          }
        />
      </View>
    </View>
    <View
      pTop={20}
      style={
        Array [
          Object {
            "paddingTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#8F959E",
                "fontFamily": "Inter-Regular",
                "fontSize": 13,
                "lineHeight": 14,
              },
            ]
          }
        >
          Password
        </Text>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderBottomColor": "#E7E8EA",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "width": "100%",
              },
            ]
          }
        >
          <TextInput
            autoCapitalize="none"
            keyboardType="default"
            onChangeText={[Function]}
            placeholder="Enter your password"
            secureTextEntry={true}
            style={
              Array [
                Object {
                  "color": "#1D1E20",
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "fontFamily": "Inter-Medium",
                  "fontSize": 15,
                  "lineHeight": 16,
                  "paddingLeft": 0,
                  "paddingRight": 5,
                  "paddingVertical": 15,
                },
              ]
            }
            type="password"
            underlineColorAndroid="transparent"
            value=""
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/icons/eye-off.png",
                }
              }
              style={
                Array [
                  Object {},
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <Text
      mTop={30}
      style={
        Array [
          Object {
            "color": "#EA4335",
            "fontFamily": "Inter-Regular",
            "marginTop": 30,
            "textAlign": "right",
          },
        ]
      }
      textAlign="right"
    >
      Forgot password?
    </Text>
    <View
      mTop={42}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 42,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "textAlign": "left",
            },
          ]
        }
      >
        Remember me
      </Text>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(52, 199, 89, 1)",
            "borderRadius": 30,
            "height": 18,
            "width": 32,
          }
        }
        testID={null}
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "center",
              "left": 8,
              "width": 32,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(255, 255, 255, 1)",
                "borderColor": "rgba(100, 100, 100, 1)",
                "borderRadius": 8,
                "borderWidth": 0,
                "height": 16,
                "justifyContent": "center",
                "width": 16,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>,
  <View
    style={
      Array [
        Object {
          "justifyContent": "flex-end",
        },
      ]
    }
  >
    <View
      pBottom={25}
      pHorizontal={25}
      style={
        Array [
          Object {
            "flexDirection": "row",
            "paddingBottom": 25,
            "paddingHorizontal": 25,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#8F959E",
              "fontFamily": "Inter-Regular",
              "fontSize": 13,
              "lineHeight": 18,
              "textAlign": "center",
            },
          ]
        }
        type="confirmAgree"
      >
        By connecting your account confirm that you agree with our 
        <Text
          style={
            Array [
              Object {
                "color": "#1D1E20",
                "fontFamily": "Inter-SemiBold",
                "fontSize": 13,
                "lineHeight": 16.5,
                "textAlign": "center",
              },
            ]
          }
          type="termAndCondition"
        >
          Term and Condition
        </Text>
      </Text>
    </View>
    <View
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#3D93F8",
            "borderColor": "transparent",
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 75,
            "paddingTop": 20,
            "width": "auto",
          },
        ]
      }
    >
      <Text
        disabled={false}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Inter-Medium",
              "fontSize": 17,
            },
          ]
        }
        type="bottom"
      >
        Login
      </Text>
    </View>
  </View>,
]
`;
